### ISSUE TITLE
Simultaneous requests for reserved JS property

### ISSUE THREAD

**Issue by jomo** [crafatar/crafatar#145]
Requests such as `/avatars/constructor` produce the error:
```
uncaughtException TypeError: requests[type][userId].push is not a function
    at push_request (lib/helpers.js:125:26)
    at store_images (lib/helpers.js:157:5)
    at lib/helpers.js:234:9
    at Command.callback (lib/cache.js:137:5)
    at RedisClient.return_reply (node_modules/redis/index.js:610:25)
    at RedisClient.<anonymous> (node_modules/redis/index.js:284:18)
    at doNTCallback0 (node.js:408:9)
    at process._tickCallback (node.js:337:13)
```

**Comment by jomo** [comment]
Fixed in c8d74d47be813dd069aff506664505ed67e7c710

**Comment by jomo** [comment]
Let me explain the problem:

We [store](https://github.com/crafatar/crafatar/blob/5e1e7f37011221a2912d1cf6d0334202a1f4394f/lib/helpers.js#L113-L115) simultaneous\* [requests](https://github.com/crafatar/crafatar/blob/5e1e7f37011221a2912d1cf6d0334202a1f4394f/lib/helpers.js#L155-L160) in an Object.
The `userId` is used as the key name for the array containing the requests:

```js
function push_request(userId, type, fun) {
  if (!requests[type][userId]) {
    requests[type][userId] = [];
  }
  requests[type][userId].push(fun);
}
```

Now in case `userId` is an [`Object.prototype` property or method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/prototype), such as `constructor`, the code above is equal to:

```js
function push_request(userId, type, fun) {
  if (!requests[type].constructor) {
    requests[type].constructor = [];
  }
  requests[type].constructor.push(fun);
}
```
Of course the constructor is not falsy, so it's not overwritten with `[]` and `requests[type].constructor.push(fun)` throws the error.

----

The fix for this is to prefix each key with `!` because overwriting prototype properties is obviously a bad idea.

----- 
_\* In fact we store every request here, but it's only useful for simultaneous requests._
